app/src/views/AccountMyFormsDref/DownloadImportTemplateButton/DownloadImportTemplateModal/useImportTemplateSchema.ts

```diff
@@ -0,0 +1,767 @@
+import useNationalSociety from '#hooks/domain/useNationalSociety';
+import { type TemplateSchema } from '#utils/importTemplate';
+import { type DrefRequestBody } from '#views/DrefApplicationForm/schema';
+
+function useImportTemplateSchema() {
+    const nationalSocieties = useNationalSociety();
+    const countries = useCountry();
+    const disasterTypes = useDisasterTypes();
+
+    const {
+        dref_planned_intervention_title,
+        dref_national_society_action_title,
+        dref_identified_need_title,
+        dref_dref_onset_type,
+        dref_dref_disaster_category,
+    } = useGlobalEnums();
+
+    const optionsMap = useMemo(() => ({
+        __boolean: [
+            {
+                key: true,
+                label: 'Yes',
+            },
+            {
+                key: false,
+                label: 'No',
+            },
+        ],
+        national_society: nationalSocieties.map(
+            ({ id, society_name }) => ({ key: id, label: society_name }),
+        ),
+        country: countries?.map(
+            ({ id, name }) => ({ key: id, label: name }),
+        ),
+        disaster_type: disasterTypes?.map(
+            ({ id, name }) => ({ key: id, label: name }),
+        ) ?? [],
+        type_of_onset: dref_dref_onset_type?.map(
+            ({ key, value }) => ({ key, label: value }),
+        ) ?? [],
+        disaster_category: dref_dref_disaster_category?.map(
+            ({ key, value }) => ({ key, label: value }),
+        ) ?? [],
+        planned_interventions: dref_planned_intervention_title?.map(
+            ({ key, value }) => ({ key, label: value }),
+        ) ?? [],
+        source_information: [
+            { key: 'source__0', label: 'Source #1' },
+            { key: 'source__1', label: 'Source #2' },
+            { key: 'source__2', label: 'Source #3' },
+        ],
+        planned_interventions__indicators: [
+            { key: 'indicator__0', label: 'Indicator #1' },
+            { key: 'indicator__1', label: 'Indicator #2' },
+            { key: 'indicator__2', label: 'Indicator #3' },
+        ],
+        risk_security: [
+            { key: 'risk__0', label: 'Risk #1' },
+            { key: 'risk__1', label: 'Risk #2' },
+            { key: 'risk__2', label: 'Risk #3' },
+        ],
+        national_society_actions: dref_national_society_action_title?.map(
+            ({ key, value }) => ({ key, label: value }),
+        ) ?? [],
+        identified_needs: dref_identified_need_title?.map(
+            ({ key, value }) => ({ key, label: value }),
+        ) ?? [],
+    }), [
+        countries,
+        disasterTypes,
+        nationalSocieties,
+        dref_planned_intervention_title,
+        dref_national_society_action_title,
+        dref_identified_need_title,
+        dref_dref_onset_type,
+        dref_dref_disaster_category,
+    ]);
+
+    const drefFormSchema: TemplateSchema<DrefRequestBody, typeof optionsMap> = useMemo(() => ({
```

Comment:
@frozenhelium we need to check if we can change the fields from a mapping to an object.

We should be able to also add "Adhoc" header rows and it might no be possible with the current structure. :crying_cat_face: